### PR TITLE
Fix submitting and requesting progress reports

### LINKED ISSUE #191: Submitting / requesting progress reports and saving files
Only the additional pdf (if there is one) should be saved to disk when submitting a progress report.

No file should be saved to disk when requesting a progress report.

### PR DESCRIPTION
**Pull request by hettlage**
* Do not create and save the progress report when submitting; only save the additional pdf (if there is one).

* Do not save any files when requesting a progress report.

- - -
This change is [Reviewable](https://reviewable.io/reviews/saltastroops/salt-api/193)

### CODE REVIEW

**Review by mofokeng-chaka — CHANGES_REQUESTED** [review]
Some progress report integration tests are now failing.

> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/saltastroops/salt-api/193#-:-NPRYZr81_3wvfnQEBFN:b-7mhuk5)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @hettlage)

**Review by hettlage — COMMENTED** [review]
Thanks for pointing this out. Should be fixed now.

> *[Reviewable](https://reviewable.io/reviews/saltastroops/salt-api/193#-:-NPTep03DSARdeseD6EV:bejrlh3)* status: 1 of 5 files reviewed, all discussions resolved (waiting on @mofokeng-chaka)

**Review by mofokeng-chaka — CHANGES_REQUESTED** [review]
Thanks. I approve this PR but before I can't merge this branch because it has conflicts. Could you please have a look at them?

> Reviewed 4 of 4 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/saltastroops/salt-api/193#-:-NPWi1Lv4KeGKKu_tgSt:b-7r9bxy)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @hettlage)

**Pull request merged**
